File: GithubCopilot_BP/CWE-119/119_2b_BP.py
import sys

def main():
    # Define a sample array for demonstration
    sample_array = [10, 20, 30, 40, 50]

    # Step 1: Input Validation
    if len(sys.argv) != 2:
        print("Usage: python script.py <index>")
        sys.exit(1)  # Exit the program indicating error

    try:
        # Step 2: Convert command line argument to integer
        index = int(sys.argv[1])
    except ValueError:
        print("Invalid input: Please enter a valid integer.")
        sys.exit(1)  # Exit the program indicating error

    # Step 3: Limiting Access - Check if the integer is within the range of the array
    if 0 <= index < len(sample_array):
        # If within range, print the element at that index
        print(sample_array[index])
    else:
        # If not within range, print "invalid input"
        print("Invalid input: Index out of range.")

    # Note: Python automatically handles memory cleanup for you, so there's no need for explicit memory cleanup in this simple script.

if __name__ == "__main__":
    main()

File: GithubCopilot_BP_Cpp/CWE-119/119_2b_BP.cpp
#include <iostream>
#include <string>
#include <stdexcept>

int main(int argc, char* argv[]) {
    // Define an array of integers
    int arr[] = {10, 20, 30, 40, 50};
    int arrSize = sizeof(arr) / sizeof(arr[0]);

    // Check if the correct number of arguments is provided
    if (argc != 2) {
        std::cerr << "Usage: " << argv[0] << " <index>" << std::endl;
        return 1;
    }

    try {
        // Convert the command-line argument to an integer
        int index = std::stoi(argv[1]);

        // Validate the index
        if (index >= 0 && index < arrSize) {
            std::cout << "Element at index " << index << " is " << arr[index] << std::endl;
        } else {
            std::cerr << "invalid input" << std::endl;
        }
    } catch (const std::invalid_argument& e) {
        std::cerr << "invalid input" << std::endl;
    } catch (const std::out_of_range& e) {
        std::cerr << "invalid input" << std::endl;
    }

    return 0;
}